Agrego timeout para la conexión

sjdalessandro opened the pull request.

sjdalessandro committed:
Agregar timeout de conexión

sjdalessandro committed:
Reparar test upload

sjdalessandro committed:
Utilizar constante SOCKET_TIMEOUT

sjdalessandro committed:
Cambiar comentario

MegaRedHand committed:
Receiver ignora los paquetes repetidos

MegaRedHand commented on src/upload.py:
```diff
@@ -31,7 +31,7 @@ def send_request(s, addr, msg):
 def upload(server_address, src, name):
     s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    s.settimeout(constant.RETRY_DELAY)
+    s.settimeout(constant.SOCKET_TIMEOUT)
```
Debería ser `CONNECTION_TIMEOUT`. En el cliente en vez de usar timers uso el timeout del socket para saber cuándo considerar que se perdió el paquete, que es lo mismo que hace `Connection` en `timed_out`.

sjdalessandro committed:
Hacer ACK en Receiver

MegaRedHand committed:
Receiver ignora paquetes viejos

No manda ACKs para paquetes ya recibidos previamente, salvo por el último recibido.

MegaRedHand committed:
Varios fixes

- corrijo timers en Connection (ahora usa monotonic)
- Sender ignora ACKs duplicados luego de reenviar paquetes (por un
  tamaño de ventana)
- corrijo timeout en el cliente (estaba duplicando la recepción de
  paquetes)
- cambio el formato de algunos logs

MegaRedHand committed:
Modifico llenado de files en pruebas

para mayor exactitud en el tamaño del archivo

MegaRedHand committed:
El buffer ya funciona

Se estaba moviendo más de lo necesario cuando llegaba al final del archivo, y sacaba del buffer paquetes no confirmados. Entonces al reenviar paquetes, enviaba paquetes de menos.

MegaRedHand committed:
Agrego constante y break

- agrego constante con el tamaño del header
- faltaba un break en protocol

sjdalessandro committed:
Cambiar funcionamiento de la ventana de gbn (#19)

* Cambiar funcionamiento de la ventana de gbn

* Corrijo error en el inicio de la conexión

El cliente repetía el número de secuencia del servidor, por lo que si se
perdía el paquete 1, se quedaba en el loop

* Unifico partes de upload y download

* Aplico formato

* Agrego chequeo finalización de conexión al cliente

Co-authored-by: Tomas Gruner <47506558+MegaRedHand@users.noreply.github.com>
